docs/developer-docs/latest/update-migration-guides/migration-guides/v4/code-migration.md

```diff
@@ -1,16 +1,42 @@
+:::
+
+For each topic covered, this guide is designed to:
+
+* help you understand the main differences between Strapi v3 and v4,
+* help you resolve breaking changes by migrating built-in code to Strapi v4,
+* and, whenever possible, offer some clues and starting points on how to migrate custom code.
+
+The following topics are covered by this guide, and you can either click on a specific topic to jump to the dedicated migration documentation or click on a scope to read a more general introduction for the back end or the front end migration:
+
+| Scope                    | Topic                                                                                                                                      |
+| ------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------ |
+| [Back end code migration](/developer-docs/latest/update-migration-guides/migration-guides/v4/code/backend.md)  | <ul><li>[Routes](/developer-docs/latest/update-migration-guides/migration-guides/v4/code/backend/routes.md)</li><li>[Policies](/developer-docs/latest/update-migration-guides/migration-guides/v4/code/backend/policies.md)</li><li>[Route middlewares](/developer-docs/latest/update-migration-guides/migration-guides/v4/code/backend/route-middlewares.md)</li><li>[Controllers](/developer-docs/latest/update-migration-guides/migration-guides/v4/code/backend/controllers.md)</li><li>[Services](/developer-docs/latest/update-migration-guides/migration-guides/v4/code/backend/services.md)</li><li>[Dependencies](/developer-docs/latest/update-migration-guides/migration-guides/v4/code/dependencies.md)</li><li>[GraphQL resolvers](/developer-docs/latest/update-migration-guides/migration-guides/v4/code/backend/graphql.md)</li></ul> |
+| [Front end code migration](/developer-docs/latest/update-migration-guides/migration-guides/v4/code/frontend.md) | <ul><li>[WYSIWYG customizations](/developer-docs/latest/update-migration-guides/migration-guides/v4/code/frontend/wysiwyg.md)</li><li>[Translations](/developer-docs/latest/update-migration-guides/migration-guides/v4/code/frontend/translations.md)</li><li>[Webpack configuration](/developer-docs/latest/update-migration-guides/migration-guides/v4/code/frontend/webpack.md)</li><li>[Theme customizations](/developer-docs/latest/update-migration-guides/migration-guides/v4/code/frontend/theming.md)</li><li>[Calls to the `strapi` global variable](/developer-docs/latest/update-migration-guides/migration-guides/v4/code/frontend/strapi-global.md)</li></ul> |
+
+::: note
+The following topics are not extensively covered in this code migration guide, but Strapi v4 also introduces:
 
-Migrating a Strapi application from v3.6.8 to v4.0.x consists in the following steps:
+* a new [file structure](/developer-docs/latest/setup-deployment-guides/file-structure.md) for the project,
```

Comment:
Weren't we planning to do a file structure migration guide? Or looping in the codemods?